tools/highlight-files/highlight-files.ts

```diff
@@ -25,21 +26,48 @@ function getBazelActionArguments() {
   return args;
 }
 
+function _detectAndHighlightRegionBlocks(parsed:
+                                           { contents: string, regions: { [p: string]: string } },
+                                         basePath: string,
+                                         outDir: string,
+                                         fileExtension: string) {
+  for (const [regionName, regionSnippet] of Object.entries(parsed.regions)) {
+    // Create files for each found region
+    if (!regionName) {
+      continue;
+    }
+    const highlightedRegion = highlightCodeBlock(regionSnippet, fileExtension);
+    // Convert "my-component-example.ts" into "my-component-example_region-ts.html"
+    const regionBaseOutputPath = basePath.replace(`.${fileExtension}`,
```

Comment:
I ask about that above, because someone could pass a file extension that doesn't match and you'd overwrite the original file

Reply:
I figured since it was already computed it didn't make sense to do it again. I hadn't considered this because it was supposed to be just a helper, but it's better to be safe